### PR TITLE
Make sure we start scanning at an indexed location (fixes #1399)

### LINKED ISSUE #1399: Accepting remote scan requests should be more clever
https://github.com/syncthing/syncthing-inotify/issues/22

We should make sure we have indexed everything up to the directory we are being asked to scan.

### PR DESCRIPTION
**Pull request by AudriusButkevicius**

### CODE REVIEW

**Comment by calmh** [comment]
I think this is an infinite loop if sub is an unhindered did ending in slash?

**Comment by AudriusButkevicius** [comment]
:exclamation:

**Comment by calmh** [comment]
Actually it probably isn't because the first filepath.Dir removes the slash and then you're in business again.

**Comment by AudriusButkevicius** [comment]
But this breaks if sub == "/"

**Comment by AudriusButkevicius** [comment]
Should be fine now.

File: internal/model/model.go [hunk 1]
```diff
@@ -1162,6 +1163,19 @@ func (m *Model) ScanFolderSub(folder, sub string) error {
 
 	_ = ignores.Load(filepath.Join(folderCfg.Path, ".stignore")) // Ignore error, there might not be an .stignore
 
+	// Required to make sure that we start indexing at a directory we already
```
Comment (calmh) [inline internal/model/model.go]: > calmh puts on his robe and pedantic wizard hat

**we're** already

File: internal/model/model.go [hunk 2]
```diff
@@ -1162,6 +1163,19 @@ func (m *Model) ScanFolderSub(folder, sub string) error {
...
+	// aware off.
+	for sub != "" {
+		_, ok = fs.Get(protocol.LocalDeviceID, sub)
+		if ok {
```
Comment (calmh) [inline internal/model/model.go]: good opportunity for `if _, ok = ...; ok {`

**Comment by calmh** [comment]
LGTM :)

File: internal/model/model.go (hunk 2 above)
Comment (AudriusButkevicius) [inline internal/model/model.go]: Not really, ok is already on the stack.

**Pull request merged**